x-pack/plugin/diskbbq/src/internalClusterTest/java/org/elasticsearch/license/DiskBBQLicensingIT.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.license;
+
+import org.apache.lucene.tests.util.LuceneTestCase;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.internal.XPackLicenseStatus;
+import org.elasticsearch.plugins.Plugin;
+import org.elasticsearch.search.vectors.KnnSearchBuilder;
+import org.elasticsearch.test.ESIntegTestCase;
+import org.elasticsearch.xcontent.XContentType;
+import org.elasticsearch.xpack.diskbbq.LocalStateDiskBBQ;
+import org.junit.Before;
+
+import java.util.Collection;
+import java.util.List;
+
+@ESIntegTestCase.ClusterScope(scope = ESIntegTestCase.Scope.TEST, numDataNodes = 1, numClientNodes = 0, supportsDedicatedMasters = false)
+@LuceneTestCase.SuppressCodecs("*")
+public class DiskBBQLicensingIT extends ESIntegTestCase {
+
+    @Before
+    public void resetLicensing() {
+        enableLicensing();
+
+        ensureStableCluster(1);
+        ensureYellow();
+    }
+
+    @Override
+    protected Settings nodeSettings(int nodeOrdinal, Settings otherSettings) {
+        Settings.Builder settings = Settings.builder().put(super.nodeSettings(nodeOrdinal, otherSettings));
+        settings.put(LicenseSettings.SELF_GENERATED_LICENSE_TYPE.getKey(), "trial");
+        return settings.build();
+    }
+
+    public void testCreateDiskBBQIndexRestricted() {
+        disableLicensing(randomInvalidLicenseType());
```

Comment:
What does it mean disabling invalid license? 
If the license type is already invalid, doesn't support DiskBBQ anyway, so why also mark it as inactive?

Reply:
good callout, let me adjust these tests. we should test with NO license and with a random invalid license.